src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -226,7 +226,7 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
                     _success = false;
                 }
 
-                await ShowHttpSourcesError();
+                _success &= await ShowHttpSourcesError();
 
```

Comment:
not familiar with the code, naive question, why doesn't all this stop in the case success is false? it seems like old style HRESULT (but without testing) rather than throwing on failure.

Reply:
@danmoseley - The reason is that we still need to write the error to the assets file so that it can be displayed in Visual Studio. However, there's probably more work we can avoid doing when encountering an http source. We will work on additional improvements for .NET 10.

Reply:
Our goal is that we explicitly want to report all lint errors prior to exiting. 

These types of include things like, missing a CPM version for a package, using VersionOverride while it's explicitly disallowed etc. This allows the user to fix all the errors prior to rerunning.
The http errors check is another one of these errors. It's not really combinable in the same way the CPM errors are, it is a lint error, so it follows that pattern.

Reply:
gotcha thanks, not attempting to suggest I know this like you guys do. :)